pkg/services/authn/grpcutils/ns_authorizer.go

```diff
@@ -0,0 +1,30 @@
+package grpcutils
+
+import (
+	authzlib "github.com/grafana/authlib/authz"
+	"github.com/grafana/authlib/claims"
+
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+func NewNamespaceAuthorizer(cfg *setting.Cfg) authzlib.AuthorizeFunc {
+	var na authzlib.NamespaceAccessChecker
+
+	if cfg.StackID != "" {
+		na = authzlib.NewNamespaceAccessChecker(
+			claims.CloudNamespaceFormatter,
+			authzlib.WithIDTokenNamespaceAccessCheckerOption(true),
```

Comment:
If we want to allow service calls, we should probably set the `required` bool to `false`.